File: engine/engine-model/src/main/scala/net/server/channel/packet/interaction/CreateShopPlayerInteractionPacket.scala
package net.server.channel.packet.interaction

class CreateShopPlayerInteractionPacket(private var _mode: Byte, private var _createType: Byte, private var _description: String, private var _itemId: Int) extends BaseCreatePlayerInteractionPacket(_mode: Byte, _createType: Byte) {
  def description: String = _description

  def itemId: Int = _itemId
}

File: engine/engine-model/src/main/java/net/server/channel/packet/interaction/CreateShopPlayerInteractionPacket.java
package net.server.channel.packet.interaction;

public class CreateShopPlayerInteractionPacket extends BaseCreatePlayerInteractionPacket {
   private final String description;

   private final Integer itemId;

   public CreateShopPlayerInteractionPacket(Byte mode, Byte createType, String description, Integer itemId) {
      super(mode, createType);
      this.description = description;
      this.itemId = itemId;
   }

   public String description() {
      return description;
   }

   public Integer itemId() {
      return itemId;
   }
}
